components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/main/java/org/wso2/carbon/identity/api/server/action/management/v1/util/ActionMgtEndpointUtil.java

```diff
@@ -0,0 +1,166 @@
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_CODE_INVALID_ACTION_ENDPOINT_AUTHENTICATION_PROPERTIES;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_CODE_INVALID_ACTION_ENDPOINT_AUTHENTICATION_TYPE;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_CODE_INVALID_ACTION_ENDPOINT_URI;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_CODE_INVALID_ACTION_NAME;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_CODE_INVALID_ACTION_TYPE;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.ErrorMessage.ERROR_CODE_NOT_SUPPORTED_ACTION_ENDPOINT_AUTHENTICATION_TYPE;
+import static org.wso2.carbon.identity.api.server.action.management.v1.constants.ActionMgtEndpointConstants.PATH_CONSTANT;
+import static org.wso2.carbon.identity.api.server.common.Constants.ERROR_CODE_DELIMITER;
+
+/**
+ * Utility class for Action management endpoint.
+ */
+public class ActionMgtEndpointUtil {
+
+    private static final Log LOG = LogFactory.getLog(ActionMgtEndpointUtil.class);
+    private static final String ACTION_TYPE_LINK_FORMAT = Constants.V1_API_PATH_COMPONENT + ACTION_PATH_COMPONENT
+            + PATH_CONSTANT;
+
+    public static String getValidatedActionType(String actionType) {
+
+        for (ActionMgtEndpointConstants.ActionTypes type: ActionMgtEndpointConstants.ActionTypes.values()) {
+
+            if (type.getPathParam().equals(actionType)) {
+                return type.getActionType();
+            }
+        }
+        throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_ACTION_TYPE);
+    }
+
+    public static void validateAction(ActionModel actionModel) {
+
+        if (StringUtils.isBlank(actionModel.getName())) {
+            throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_ACTION_NAME);
+        }
+        if (actionModel.getEndpoint() == null) {
+            throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_ACTION_ENDPOINT);
+        }
+        if (StringUtils.isBlank(actionModel.getEndpoint().getUri())) {
+            throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_ACTION_ENDPOINT_URI);
+        }
+        if (actionModel.getEndpoint().getAuthentication() == null) {
+            throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_ACTION_ENDPOINT_AUTHENTICATION);
+        }
+        if (StringUtils.isBlank(actionModel.getEndpoint().getAuthentication().getType().toString())) {
+            throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_ACTION_ENDPOINT_AUTHENTICATION_TYPE);
+        }
```

Comment:
Can't we handle these validations by open api spec definitions like adding required ?

Reply:
Fixed with https://github.com/wso2/identity-api-server/pull/626/commits/2dac506e1f7ffa6d587ddae09f7bf059c0f35f95